src/Take.Blip.Builder/FlowManager.cs

```diff
@@ -292,7 +297,7 @@ public async Task ProcessInputAsync(Message message, Flow flow, IContext message
                                 // Check if the state transition limit has reached (to avoid loops in the flow)
                                 if (transitions++ >= _configuration.MaxTransitionsByInput)
                                 {
-                                    throw new BuilderException($"Max state transitions of {_configuration.MaxTransitionsByInput} was reached");
+                                    throw new FlowConstructionException(_analyzeBuilderExceptions.CreateFlowConstructionExceptionMessage($"Max state transitions of {_configuration.MaxTransitionsByInput} was reached"));
```

Comment:
move formatting message to flow construction exception inside the FlowConstructionException constructor